spec/controllers/course/registrations_controller_spec.rb

```diff
@@ -0,0 +1,217 @@
+  before do
+    ActsAsTenant.current_tenant = Instance.default
+  end
+
+  let!(:user) { create(:user) }
+  let!(:course) { create(:course, status: :opened) }
+  let!(:student_request) { create(:student_registration, course: course) }
+
+  let!(:student_request2) { create(:student_registration, course: course) }
+  let!(:staff_request) { create(:teaching_assistant_registration, course: course) }
+  let!(:staff_request2) { create(:manager_registration, course: course) }
+
+  describe '#index' do
+    context 'user can manage the course' do
+      before do
+        sign_in(create(:administrator))
+      end
+
+      subject { get :index, course_id: course.id }
+      it 'renders index template' do
+        subject
+        expect(response).to render_template(:index)
+        expect(assigns(:student_requests)).to contain_exactly(student_request, student_request2)
+        expect(assigns(:staff_requests)).to contain_exactly(staff_request, staff_request2)
+      end
+    end
+  end
+
+  describe '#create' do
+    subject { post :create, course_id: course, registration: student_request.as_json }
+
+    context 'user is signed in' do
+      before do
+        sign_in(user)
+      end
+
+      context 'course is open' do
+        it 'creates a new request after first attempt' do
+          expect { subject }.to change(Course::Registration, :count).by(1)
+          expect(response).to redirect_to(course_path(course))
+          expect(flash[:notice]).
+            to match(I18n.t('course.registrations.create.request_format') %
+                       { role: I18n.t("course.registrations.role.#{student_request.role}") })
+        end
+      end
+
+      context 'course is closed' do
+        before do
+          course.status = :published
+          course.save!
+        end
+        it 'redirects to course page' do
+          subject
+          expect(response).to redirect_to(course_path(course))
+          expect(flash[:error]).to match(I18n.t('course.registrations.create.course_not_open'))
+        end
+      end
+
+      context 'already registered' do
+        before do
+          create(:course_student, course: course, user: user)
+        end
+
+        it 'redirects to course page' do
+          subject
+          expect(response).to redirect_to(course_path(course))
+          expect(flash[:error]).
+            to match(I18n.t('course.registrations.create.already_registered_format') %
+                       { role: I18n.t('course.registrations.role.student') })
+
+        end
+      end
+    end
+
+    context 'user is not signed in' do
+      it 'redirects user to sign in page' do
+        subject
+        expect(response).to redirect_to(new_user_session_path)
+      end
+    end
+  end
+
+  describe '#update_multiple' do
+    before do
+      sign_in(create(:administrator))
+    end
+
+    context 'approve selected staff' do
+      subject { post :update_multiple, course_id: course.id, approve_selected: 'staff',
```

Comment:
Avoid using {...} for multi-line blocks.<br>Block body expression is on the same line as the block start.